numpyro/contrib/tfp/distributions.py

```diff
@@ -90,7 +90,23 @@ def _transform_to_bijector_constraint(constraint):
     return BijectorTransform(constraint.bijector)
 
 
-class TFPDistributionMixin(NumPyroDistribution):
+_TFPDistributionMeta = type(tfd.Distribution)
+
+
+# XXX: we create this mixin class to avoid metaclass conflict between TFP and NumPyro Ditribution
+class _TFPMixinMeta(_TFPDistributionMeta, type(NumPyroDistribution)):
+    def __init__(cls, name, bases, dct):
+        # XXX: _TFPDistributionMeta.__init__ registers cls as a PyTree
+        # for some reasons, when defining metaclass of TFPDistributionMixin to be _TFPMixinMeta,
+        # TFPDistributionMixin will be registered as a PyTree 2 times, which is not allowed
+        # in JAX, so we skip registering TFPDistributionMixin as a PyTree.
```

Comment:
Wow, that must have been tricky to debug!

Reply:
Yeah, it is tricky. Especially to understand `__call__` `__new__` `__init__` logics of metaclass. >___<